utility/test/utility.cpp

```diff
@@ -176,4 +177,27 @@ SCENARIO("isHexadecimal") {
     }
 }
 
+SCENARIO("binaryCopy") {
+    // integer representations of floats and doubles have been computed using
+    // http://babbage.cs.qc.cuny.edu/IEEE-754/
+
+    float in1 = 1.23456f;
+    // We assume that a float is coded on 32bits. That assumption is made in
+    // the Parameter Framework.
+    REQUIRE(sizeof(float) == sizeof(uint32_t));
+    auto out1 = utility::binaryCopy<uint32_t>(in1);
+    CHECK(out1 ==  0x3f9e0610);
+
+    // We assume that a double is coded on 64bits. That assumption is made in
+    // the Parameter Framework.
+    uint64_t in2 = 0x408edd3c0cb3420e;
+    REQUIRE(sizeof(double) == sizeof(uint64_t));
+    auto out2 = utility::binaryCopy<double>(in2);
+    CHECK(out2 == 987.65432109876);
+
+    uint8_t in3 = 0xff;
+    auto out3 = utility::binaryCopy<int8_t>(in3);
+    CHECK(out3 == -1);
+}
+
```

Comment:
Let me sumerize my proposition:
~~~cpp
template <class V1, class V2>
checkBinaryEqual(V1 v1, V2 v2) {
    CAPTURE(V1);
    CAPTURE(V2);
    CHECK(utility::binaryCopy<V1>(v2) == v1)
    CHECK(utility::binaryCopy<V2>(v1) == v2)
}

SCENARIO("binaryCopy") {
    WITH("Integer representation computed using http://babbage.cs.qc.cuny.edu/IEEE-754/") {

        THEN("Float should coded on 32bits. That assumption is made in the Parameter Framework.") {
            REQUIRE(sizeof(float) == sizeof(uint32_t));
        }
        WHEN("Testing float <=> uint32_t conversion") {
            checkBinaryEqual<float, int32_t>(1.23456f, 0x3f9e0610);
        }
        WHEN("Testing double <=> uint64_t conversion") {
            checkBinaryEqual<double, int64_t>(0x408edd3c0cb3420e, 987.65432109876)
        }

        WHEN("Testing int8_t <=> uint8_t conversion") {
            checkBinaryEqual<int8_t, uint8_t>(-1, 0xff);
        }
}
~~~